extensions/arc/deployment/src/main/java/io/quarkus/arc/deployment/ConfigBuildStep.java

```diff
@@ -315,23 +375,108 @@ private static void collectInterfacesRec(ClassInfo current, IndexView index, Set
         }
     }
 
+    @BuildStep
+    void validateConfigMappingsInjectionPoints(
+            ValidationPhaseBuildItem validationPhase,
+            List<ConfigClassBuildItem> configClasses,
+            BuildProducer<ConfigMappingBuildItem> configMappings) {
+
+        Set<ConfigMappingBuildItem> toRegister = new HashSet<>();
+        for (InjectionPointInfo injectionPoint : validationPhase.getContext().getInjectionPoints()) {
```

Comment:
This iteration is also suboptimal. First of all, we don't need to iterate over all injection points if no `configClasses` are defined. Furthermore, we could avoid the nested loop using a similar technique as described in https://github.com/quarkusio/quarkus/pull/19965/files#r719992140/